src/app/Library/CrudPanel/Traits/ColumnsProtectedMethods.php

```diff
@@ -80,14 +80,55 @@ protected function makeSureColumnHasLabel($column)
      */
     protected function makeSureColumnHasType($column)
     {
-        $could_be_relation = isset($column['entity']) && $column['entity'] !== false;
+        // Do not alter type if it has been set by developer
+        if (isset($column['type'])) {
+            return $column;
+        }
+
+        // Set text as default column type
+        $column['type'] = 'text';
+
+        $could_be_relation = Arr::get($column, 'entity', false) !== false;
 
-        if (! isset($column['type']) && $could_be_relation) {
+        if ($could_be_relation) {
             $column['type'] = 'relationship';
         }
 
-        if (! isset($column['type'])) {
-            $column['type'] = 'text';
+        if (in_array($column['name'], $this->model->getDates())) {
+            $column['type'] = 'datetime';
+        }
+
+        if ($this->model->hasCast($column['name'])) {
+            $attributeType = $this->model->getCasts()[$column['name']];
+
+            switch ($attributeType) {
+                case 'array':
+                case 'encrypted:array':
+                case 'collection':
+                case 'encrypted:collection':
+                case 'json':
+                case 'object':
+                    $column['type'] = 'array';
+                    break;
+                case 'bool':
+                case 'boolean':
+                    $column['type'] = 'check';
```

Comment:
Note I have been slight opinionated here to pick the `check` column over the `boolean` as I like it better. If you prefer the second let me know.

Reply:
I have the same preference 😀